packages/puppeteer-core/src/common/ElementHandle.ts

```diff
@@ -412,6 +421,61 @@ export class ElementHandle<
     return this.waitForSelector(`xpath/${xpath}`, options);
   }
 
+  /**
+   * Converts the current handle to the given element type.
+   *
+   * @example
+   *
+   * ```ts
+   * const element: ElementHandle<Element> = await page.$(
+   *   '.class-name-of-anchor'
+   * );
+   * // DO NOT DISPOSE `element`, this will be always be the same handle.
+   * const anchor: ElementHandle<HTMLAnchorElement> = await element.toElement(
+   *   'a'
+   * );
+   * ```
+   *
+   * @example
+   * Garbage handling is automatically handled on error, but you can turn this off.
+   *
+   * ```ts
+   * const element: ElementHandle<Element> = await page.$(
+   *   '.class-name-of-anchor'
+   * );
+   * try {
+   *   const anchor = await element.toElement('svg', {dispose: false});
+   * } catch (err) {
+   *   // You need to handle the trash since you specified `dispose: false`
+   *   element.dispose();
+   *   throw new err();
+   * }
+   * ```
+   *
+   * @param tagName - The tag name of the desired element type.
+   * @param options - Configures the behavior of this method.
+   * @throws An error if the handle does not match. **This will dispose the
+   * current handle.** You can disable this behavior in the options.
+   */
+  async toElement<
+    K extends keyof HTMLElementTagNameMap | keyof SVGElementTagNameMap
+  >(
+    tagName: K,
+    options: ToElementOptions = {}
+  ): Promise<HandleFor<ElementFor<K>>> {
+    const {dispose = true} = options;
+    const isMatchingTagName = await this.evaluate((node, tagName) => {
+      return node.nodeName === tagName.toUpperCase();
+    }, tagName);
+    if (!isMatchingTagName) {
+      if (dispose) {
+        await this.dispose();
+      }
+      throw new Error(`Element is not a(n) \`${tagName}\` element`);
+    }
+    return this as unknown as HandleFor<ElementFor<K>>;
+  }
+
   override asElement(): ElementHandle<ElementType> | null {
```

Comment:
could we replace existing usages of `asElement` in our codebase if it gets deprecated?

Reply:
We can do this when we remove `asElement` completely. It's not that important right now.